concepts/patterns/introduction.md

```diff
@@ -0,0 +1,75 @@
+# Introduction
+
+Recall from the Fundamentals concept, an AWK program is composed of **pattern-action pairs**.
+
+```awk
+pattern1 { action1 }
+pattern2 { action2 }
+...
+```
+
+## What do we mean by "pattern"?
+
+The "pattern" is any AWK expression that results in a true or false value.
+
+## The Empty Pattern
+
+The pattern can be omitted.
+In this case, the action is performed for every record.
+
+We can print all the usernames in the passwd file.
+
+```sh
+awk -F: '{print $1}' /etc/passwd
+```
+
+## Regular Expressions
+
+AWK can do what grep can do.
```

Comment:
This assumes people know grep and can use it as a reference. Prefer something like, AWK can search for a regular expression, similar to `grep`.

If you want to make the comparison more explicit, you could be explicit: This accomplishes the same as  `grep pattern data.txt`